src/Bounds.cpp

```diff
@@ -733,6 +733,11 @@ class Bounds : public IRVisitor {
         }
     }
 
+    void visit(const Shuffle *op) {
+        // TODO: We could consider the bounds of the lanes used by the Shuffle.
```

Comment:
I think it's worth taking the union of the bounds of the argument vectors (e.g. what if they're all broadcasts of a constant, because you're using shuffle to make the vector 0 1 0 1 0 1...)